### ISSUE TITLE
History graph shows (formatted?) entity id, not entity name

### ISSUE THREAD

**Issue by gperreault** [home-assistant/frontend#12616]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

In the history graph, the color coded name shows a formatted version of the entity id rather than the entity name.

For instance, for an entity...
- named "Home Theatre Outlet: Electric Consumption [W]"
- entity id "sensor.smart_switch_6_1_electric_consumption_w"
... the displayed line item beside the color coding shows "smart switch 6 1 electric consumption w", rather than the entity name.
This _is_ working for most entities, but not this particular one for some reason.

### Describe the behavior you expected

The entity name should be displayed in the area describing the color of the entity, rather than a formatted entity id.

### Steps to reproduce the issue

1.Add history graph with entity whose id doesn't match name
2.Note that entity name is not displayed, but a formatted version of the entity id.
...

### What version of Home Assistant Core has the issue?

2022.5.1

### What was the last working version of Home Assistant Core?

2022.3

### In which browser are you experiencing the issue with?

Microsoft Edge 101.0.1210.32 (also, Safari iOS as found in iOS 15.4.1)

### Which operating system are you using to run this browser?

Windows 10 Home 19044.1645, iOS Safari (as of iOS 15.4.1)

### State of relevant entities

```txt
sensor.smart_switch_6_1_electric_consumption_w
Home Theatre Outlet: Electric Consumption [W]
28.7	state_class: measurement
meter_type: 1
meter_type_name: ELECTRIC
unit_of_measurement: W
device_class: power
friendly_name: Home Theatre Outlet: Electric Consumption [W]
```

### Problem-relevant frontend configuration

```yaml
type: grid
title: Power Consumption
cards:
  - type: entity
    entity: sensor.smart_switch_6_1_electric_consumption_w
  - type: history-graph
    entities:
      - entity: sensor.smart_switch_6_1_electric_consumption_w
    hours_to_show: 24
    refresh_interval: 0
columns: 2
square: false
```

### Additional information

![Incorrect entity name - history graph](https://user-images.githubusercontent.com/25162738/167210234-27a8e195-7908-4e9e-b1f8-d067987de97a.PNG)

**Comment by spacegaier** [comment]
I cannot see this problem in my production instance running 2022.8. Does the issue still occur for you?

**Comment by gperreault** [comment]
Agreed, it appears to be resolved now with my current 2022.8.7 instance. Both the example I posted as well as several others that previously were displaying incorrectly appear to be displayed properly now. 

I'll close this issue.
